packages/web-extension/src/walletManager/walletManager.ts

```diff
@@ -0,0 +1,337 @@
+    if (this.#isActive(props)) {
+      return;
+    }
+
+    const { walletId, chainId, accountIndex } = props;
+
+    const wallets = await firstValueFrom(this.#walletRepository.wallets$);
+    const activeWallet = wallets.find((wallet) => wallet.walletId === walletId);
+
+    if (!activeWallet) {
+      throw new InvalidArgumentError('walletId', `Wallet ${walletId} not found`);
+    }
+
+    this.#deactivateWallet();
+    this.#activeWalletProps = props;
+
+    const walletStoreId = getWalletStoreId(walletId, chainId, accountIndex);
+    const stores = this.#getStores(walletStoreId);
+
+    const witnesser = this.#buildWitnesser(activeWallet, walletId, chainId, accountIndex);
+
+    const name = this.#getWalletName(activeWallet, accountIndex);
+
+    const [wallet] = await Promise.all([
+      this.#walletFactory.create(name, { stores, witnesser }),
+      this.#managerStorage.set({
+        [this.#managerStorageKey]: props
+      })
+    ]);
+
+    this.activeWallet$.next(wallet);
+
+    this.#activeWalletId$.next(props);
+  }
+
+  /** Deactivate wallet. Wallet observable properties will emit only after a new wallet is {@link activate}ed. */
+  async deactivate(): Promise<void> {
+    this.#deactivateWallet();
+  }
+
+  /** Deactivates the active. */
+  shutdown(): void {
+    this.#deactivateWallet();
+  }
+
+  /**
+   * Deactivates the active wallet and destroy its existing store.
+   *
+   * @param walletId The walletId of the wallet to destroy.
+   */
+  async destroyData(walletId: WalletId): Promise<void> {
+    await this.#destroyWalletStores(walletId);
+  }
+
+  /**
+   * Checks if the wallet is active.
+   *
+   * @param walletProps The wallet properties to check.
+   * @private
+   */
+  #isActive(walletProps: WalletManagerActivateProps): boolean {
+    if (!this.#activeWalletProps) return false;
+
+    return (
+      this.#activeWalletProps?.walletId === walletProps.walletId &&
+      this.#activeWalletProps?.accountIndex === walletProps.accountIndex &&
+      this.#activeWalletProps?.chainId === walletProps.chainId
+    );
+  }
+
+  /** Gets store if wallet was activated previously or creates one when wallet is activated for the first time. */
+  #getStores(walletStoreName: string): storage.WalletStores {
+    let stores = this.#walletStores.get(walletStoreName);
+    if (!stores) {
+      stores = this.#storesFactory.create({ name: walletStoreName });
+      this.#walletStores.set(walletStoreName, stores);
+    }
+    return stores;
+  }
+
+  /**
+   * Destroys all stores for the given wallet id.
+   *
+   * @param walletId The wallet id to destroy.
+   * @private
+   */
+  async #destroyWalletStores(walletId: WalletId): Promise<void> {
+    if (this.#activeWalletProps?.walletId === walletId)
+      throw new InvalidArgumentError('walletId', 'Cannot destroy active wallet');
+
+    const stores = [...this.#walletStores.keys()].filter((key) => key.includes(walletId));
```

Comment:
This won't destroy the store data if it's not loaded in memory (if wallet wasn't activated since starting)

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1012/commits/17d86e545285d041a6fd47c4d48cc5b1d8897c70